pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -198,50 +198,56 @@ impl FileExt for EphemeralFile {
 
 impl BlobWriter for EphemeralFile {
     fn write_blob(&mut self, srcbuf: &[u8]) -> Result<u64, io::Error> {
-        let pos = self.size;
-
-        let mut blknum = (self.size / PAGE_SZ as u64) as u32;
-        let mut off = (pos % PAGE_SZ as u64) as usize;
+        struct Writer<'a> {
+            ephemeral_file: &'a mut EphemeralFile,
+            blknum: u32,
+            off: usize,
+        }
+        impl<'a> Writer<'a> {
+            fn new(ephemeral_file: &'a mut EphemeralFile) -> Writer<'a> {
+                Writer {
+                    blknum: (ephemeral_file.size / PAGE_SZ as u64) as u32,
+                    off: (ephemeral_file.size % PAGE_SZ as u64) as usize,
+                    ephemeral_file,
+                }
+            }
+            fn push_bytes(&mut self, src: &[u8]) -> Result<(), io::Error> {
+                let mut src_remaining = src;
```

Comment:
```suggestion
            fn push_bytes(&mut self, mut src: &[u8]) -> Result<(), io::Error> {
```

Reply:
I know, but, I like the correspondence to `dst_remaining`.